test/extended/two_node/tnf_node_replacement.go

```diff
@@ -0,0 +1,1508 @@
+func findLearnerMemberID(memberListJSON string) (string, string, error) {
+	// Parse the JSON output
+	var memberList etcdMemberListResponse
+	err := utils.UnmarshalJSON(memberListJSON, &memberList)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to parse etcd member list JSON: %v", err)
+	}
+
+	// Find learner members
+	for _, member := range memberList.Members {
+		if member.IsLearner {
+			// Convert member ID to hex string format (etcdctl expects hex format)
+			memberIDHex := fmt.Sprintf("%x", member.ID)
+			klog.V(2).Infof("Found learner member: ID=%s (hex: %s), Name=%s", fmt.Sprintf("%d", member.ID), memberIDHex, member.Name)
+			return memberIDHex, member.Name, nil
+		}
+	}
+
+	// No learner found
+	klog.V(2).Infof("No learner member found in member list")
+	return "", "", nil
+}
+
+// recoverEtcdSecretsFromBackup recreates etcd secrets from backup with retry logic
+func recoverEtcdSecretsFromBackup(testConfig *TNFTestConfig, oc *exutil.CLI) error {
+	etcdSecrets := []string{
+		testConfig.EtcdResources.PeerSecretName,
+		testConfig.EtcdResources.ServingSecretName,
+		testConfig.EtcdResources.ServingMetricsSecretName,
+	}
+
+	for _, secretName := range etcdSecrets {
+		secretFile := filepath.Join(testConfig.Execution.GlobalBackupDir, secretName+".yaml")
+		if _, err := os.Stat(secretFile); os.IsNotExist(err) {
+			klog.Warningf("Backup file for etcd secret %s not found", secretName)
+			continue
+		}
+
+		// Check if the secret already exists
+		_, err := oc.AsAdmin().Run("get").Args(secretResourceType, secretName, "-n", etcdNamespace).Output()
+		if err == nil {
+			klog.V(2).Infof("Etcd secret %s already exists, skipping recreation", secretName)
+			continue
+		}
+
+		// Retry the secret creation with etcd learner error handling
+		err = core.RetryWithOptions(func() error {
+			_, err := oc.AsAdmin().Run("create").Args("-f", secretFile).Output()
+			return err
+		}, core.RetryOptions{
+			Timeout:      fiveMinuteTimeout,
+			PollInterval: thirtySecondPollInterval,
+			MaxRetries:   10,
+			ShouldRetry:  services.IsRetryableEtcdError,
+		}, fmt.Sprintf("create etcd secret %s", secretName))
+
+		if err != nil {
+			klog.Warningf("Failed to recreate etcd secret %s after retries: %v", secretName, err)
+			continue
+		}
+		klog.V(2).Infof("Recreated etcd secret: %s", secretName)
+	}
+
+	return nil
+}
+
+// recoverBMHAndMachineFromBackup recreates BMH and Machine from backup with retry logic
+func recoverBMHAndMachineFromBackup(testConfig *TNFTestConfig, oc *exutil.CLI) error {
+
+	err := recreateBMCSecret(testConfig, oc)
+	if err != nil {
+		return fmt.Errorf("failed to recreate BMC secret: %v", err)
+	}
+
+	// Recreate Machine with retry
+	machineFile := filepath.Join(testConfig.Execution.GlobalBackupDir, testConfig.TargetNode.MachineName+"-machine.yaml")
+
+	// Check if Machine already exists
+	_, err = oc.AsAdmin().Run("get").Args(machineResourceType, testConfig.TargetNode.MachineName, "-n", machineAPINamespace).Output()
```

Comment:
Same thing about the API call here, then we can remove the custom resource type, this also has the added benefit of checking the API error for things like NotFound or just plain error

```
oc.AsAdmin().MachineConfigurationClient().MachineconfigurationV1().MachineConfigs().Get()
```